packages/devtools_app/lib/src/debugger/hover.dart

```diff
@@ -0,0 +1,153 @@
+// Copyright 2021 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/material.dart';
+
+import '../theme.dart';
+
+/// Returns the word in the [line] for the provided hover [dx] offset given
+/// the [line]'s [textStyle].
+String wordForHover(double dx, TextSpan line, TextStyle textStyle) {
+  final index = _hoverIndex(dx, line, textStyle);
+  var word = '';
+  if (index == -1) return word;
+  word = line.children[index].toPlainText();
+  word = _mergedLinkedWords(word, index, line);
+  return word;
+}
+
+/// Merges words linked by a `.`.
+///
+/// For example, hovering over `bar` in `foo.bar.baz` would return
+/// `foo.bar.baz`.
+String _mergedLinkedWords(String word, int index, TextSpan line) {
+  var left = index - 1;
```

Comment:
how is the index = 1 case handled? If we are hovering over bar in foo.bar, don't we want to evaluate foo.bar? This looks like we only merge the words if there are two previous values. We should be able to change this to chain previous values one at a time instead of 2 at a time.

Reply:
We want to evaluate all words linked by a `.`. This is consistent with Chrome's behavior.

index in this case would point to `bar` and `foo` would be merged. There are test cases covering this.

Reply:
I updated the logic and the tests. If we hover over `bar` in `foo.bar.baz` we want to evaluate `foo.bar` to be consistent with Chrome. There will always be at least two items because one of the items will be `.`. Otherwise it won't be linked.